### ISSUE TITLE
Ensure Red Hat GPG Key Installed

### ISSUE THREAD

**Issue by chanaka-jostle** [OpenSCAP/scap-security-guide#1562]
Hello,

When I run the following it complains that I am missing the Red Hat GPG key. However, it is installed on my system. 

oscap xccdf eval --profile xccdf_org.ssgproject.content_profile_rht-ccp --results-arf /tmp/arf.xml --report /tmp/X-cpp-report.html /usr/share/xml/scap/ssg/content/ssg-centos7-ds.xml

[root@redacted ssh]# rpm -q gpg-pubkey --qf '%{name}-%{version}-%{release} --> %{summary}\n' 
gpg-pubkey-*** --> gpg(CentOS-7 Key (CentOS 7 Official Signing Key) <security@centos.org>)
gpg-pubkey-***--> gpg(Red Hat, Inc. (release key 2) <security@redhat.com>)

**Comment by shawndwells** [comment]
Hm. The relevant code:
https://github.com/OpenSCAP/scap-security-guide/blob/master/shared/oval/ensure_redhat_gpgkey_installed.xml#L21#L30

`````
      <criteria comment="CentOS Vendor Keys" operator="OR">
        <criteria comment="CentOS Installed" operator="OR">
          <extend_definition comment="CentOS6 installed" definition_ref="installed_OS_is_centos6" />
          <extend_definition comment="CentOS7 installed" definition_ref="installed_OS_is_centos7" />
        </criteria>
        <criterion comment="package gpg-pubkey-f4a80eb5-53a7ff4b is installed"
        test_ref="test_package_gpgkey-f4a80eb5-53a7ff4b_installed" />
        <criterion comment="package gpg-pubkey-c105b9de-4e0fd3a3 is installed"
        test_ref="test_package_gpgkey-c105b9de-4e0fd3a3_installed" />
      </criteria>
`````

Can you send the non-shadowed ``rpm -q gpg-key`` output?

e.g.
`````
$ rpm -q gpg-pubkey --qf '%{name}-%{version}-%{release} --> %{summary}\n'
gpg-pubkey-fd431d51-4ae0493b --> gpg(Red Hat, Inc. (release key 2) <security@redhat.com>)
gpg-pubkey-2fa658e0-45700c69 --> gpg(Red Hat, Inc. (auxiliary key) <security@redhat.com>)
`````

**Comment by konstruktoid** [comment]
Verified on Centos 7 using Ansible. Tested with https://github.com/konstruktoid/ansible-role-hardening and the report is present at https://github.com/konstruktoid/ansible-role-hardening/blob/master/centos7_stig-report.html.
Using https://github.com/konstruktoid/ansible-role-hardening/blob/master/defaults/main.yml#L3, https://access.redhat.com/security/team/key/

```
[vagrant@centos ~]$ rpm -q gpg-pubkey --qf '%{name}-%{version}-%{release} --> %{summary}\n'
gpg-pubkey-f4a80eb5-53a7ff4b --> gpg(CentOS-7 Key (CentOS 7 Official Signing Key) <security@centos.org>)
gpg-pubkey-fd431d51-4ae0493b --> gpg(Red Hat, Inc. (release key 2) <security@redhat.com>)
gpg-pubkey-37017186-45761324 --> gpg(Red Hat, Inc. (release key) <security@redhat.com>)
gpg-pubkey-352c64e5-52ae6884 --> gpg(Fedora EPEL (7) <epel@fedoraproject.org>)
```

**Comment by redhatrises** [comment]
With CentOS, the Red Hat GPG key is not expected to be imported. It should only be the CentOS GPG key. I ran this off of the master with no issue.

@konstruktoid it looks like you are using the Red Hat content rather than the CentOS content. The CCEs you have in your ansible code map to Red Hat content and not CentOS.

**Comment by konstruktoid** [comment]
@redhatrises hi and thanks for the reply. I thought the profiles present had been adapted to the distribution in the name since both CentOS points to RHEL.

```
$ oscap info /usr/share/xml/scap/ssg/content/ssg-centos7-ds.xml
Document type: Source Data Stream
Imported: 2015-12-08T14:13:51

Stream: scap_org.open-scap_datastream_from_xccdf_ssg-rhel7-xccdf-1.2.xml
Generated: (null)
Version: 1.2
Checklists:
	Ref-Id: scap_org.open-scap_cref_ssg-rhel7-xccdf-1.2.xml
		Status: draft
		Generated: 2015-12-08
		Resolved: true
		Profiles:
			xccdf_org.ssgproject.content_profile_standard
			xccdf_org.ssgproject.content_profile_pci-dss
			xccdf_org.ssgproject.content_profile_rht-ccp
			xccdf_org.ssgproject.content_profile_common
			xccdf_org.ssgproject.content_profile_stig-rhel7-server-upstream
		Referenced check files:
			ssg-rhel7-oval.xml
				system: http://oval.mitre.org/XMLSchema/oval-definitions-5
Checks:
	Ref-Id: scap_org.open-scap_cref_ssg-rhel7-oval.xml
	Ref-Id: scap_org.open-scap_cref_output--ssg-rhel7-cpe-oval.xml
	Ref-Id: scap_org.open-scap_cref_output--ssg-rhel7-oval.xml
Dictionaries:
	Ref-Id: scap_org.open-scap_cref_output--ssg-rhel7-cpe-dictionary.xml
$ oscap info /usr/share/xml/scap/ssg/content/ssg-centos6-ds.xml
Document type: Source Data Stream
Imported: 2015-12-08T14:14:30

Stream: scap_org.open-scap_datastream_from_xccdf_ssg-rhel6-xccdf-1.2.xml
Generated: (null)
Version: 1.2
Checklists:
	Ref-Id: scap_org.open-scap_cref_ssg-rhel6-xccdf-1.2.xml
		Status: draft
		Generated: 2015-12-08
		Resolved: true
		Profiles:
			xccdf_org.ssgproject.content_profile_CS2
			xccdf_org.ssgproject.content_profile_common
			xccdf_org.ssgproject.content_profile_server
			xccdf_org.ssgproject.content_profile_stig-rhel6-server-upstream
			xccdf_org.ssgproject.content_profile_usgcb-rhel6-server
			xccdf_org.ssgproject.content_profile_rht-ccp
			xccdf_org.ssgproject.content_profile_CSCF-RHEL6-MLS
			xccdf_org.ssgproject.content_profile_C2S
			xccdf_org.ssgproject.content_profile_pci-dss
		Referenced check files:
			ssg-rhel6-oval.xml
				system: http://oval.mitre.org/XMLSchema/oval-definitions-5
Checks:
	Ref-Id: scap_org.open-scap_cref_ssg-rhel6-oval.xml
	Ref-Id: scap_org.open-scap_cref_output--ssg-rhel6-cpe-oval.xml
	Ref-Id: scap_org.open-scap_cref_output--ssg-rhel6-oval.xml
Dictionaries:
	Ref-Id: scap_org.open-scap_cref_output--ssg-rhel6-cpe-dictionary.xml
```

```
openscap-scanner-1.2.9-5.el7_2.x86_64
scap-security-guide-0.1.25-3.el7.centos.0.1.noarch
openscap-1.2.9-5.el7_2.x86_64
```

`CentOS Linux release 7.2.1511 (Core)`

**Comment by konstruktoid** [comment]
Continuation from previous comment:
```
[vagrant@centos ~]$ sudo oscap xccdf eval --profile xccdf_org.ssgproject.content_profile_pci-dss --results-arf centos7_stig-arf.xml --report centos7_stig-report.html /usr/share/xml/scap/ssg/content/ssg-centos7-ds.xml
Title   Ensure Red Hat GPG Key Installed
Rule    xccdf_org.ssgproject.content_rule_ensure_redhat_gpgkey_installed
Ident   CCE-26957-1
Result  fail

Title   Ensure gpgcheck Enabled In Main Yum Configuration
Rule    xccdf_org.ssgproject.content_rule_ensure_gpgcheck_globally_activated
Ident   CCE-26989-4
Result  pass

Title   Ensure gpgcheck Enabled For All Yum Package Repositories
Rule    xccdf_org.ssgproject.content_rule_ensure_gpgcheck_never_disabled
Ident   CCE-26876-3
Result  pass
...
```

**Comment by redhatrises** [comment]
Thanks for providing the `scap-security-guide` version info @konstruktoid. 

First off, `scap-security-guide-0.1.25-3.el7.centos.0.1.noarch` actually does not have proper CentOS support/fixes in it. The version that should be released with `CentOS 7.3` should bring the CentOS into better alignment. 

Second, while the check says "Ensure Red Hat GPG Key Installed", don't install the Red Hat GPG Keys on CentOS. This rule is really asking you to install the distribution's GPG Key, so if running Red Hat, install the Red Hat GPG key. If running CentOS, install the CentOS GPG key.

**Comment by redhatrises** [comment]
Closing this issue as this has already been fixed upstream and is more of a downstream issue.
